dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLKernelBuilderExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json;
+using Microsoft.Azure.Cosmos;
+using Microsoft.SemanticKernel.Data;
+
+namespace Microsoft.SemanticKernel.Connectors.AzureCosmosDBNoSQL;
+
+/// <summary>
+/// Extension methods to register Azure CosmosDB NoSQL <see cref="IVectorStore"/> instances on the <see cref="IKernelBuilder"/>.
+/// </summary>
+public static class AzureCosmosDBNoSQLKernelBuilderExtensions
+{
+    /// <summary>
+    /// Register a Azure CosmosDB NoSQL <see cref="IVectorStore"/> with the specified service ID
+    /// and where the Azure CosmosDB NoSQL <see cref="Database"/> is retrieved from the dependency injection container.
+    /// </summary>
+    /// <param name="builder">The builder to register the <see cref="IVectorStore"/> on.</param>
+    /// <param name="options">Optional options to further configure the <see cref="IVectorStore"/>.</param>
+    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    /// <returns>The kernel builder.</returns>
+    public static IKernelBuilder AddAzureCosmosDBNoSQLVectorStore(
+        this IKernelBuilder builder,
+        AzureCosmosDBNoSQLVectorStoreOptions? options = default,
+        string? serviceId = default)
+    {
+        builder.Services.AddAzureCosmosDBNoSQLVectorStore(options, serviceId);
+        return builder;
+    }
+
+    /// <summary>
+    /// Register a Azure CosmosDB NoSQL <see cref="IVectorStore"/> with the specified service ID
+    /// and where the Azure CosmosDB NoSQL <see cref="Database"/> is constructed using the provided <paramref name="connectionString"/> and <paramref name="databaseName"/>.
+    /// </summary>
+    /// <param name="builder">The builder to register the <see cref="IVectorStore"/> on.</param>
+    /// <param name="connectionString">Connection string required to connect to Azure CosmosDB NoSQL.</param>
+    /// <param name="databaseName">Database name for Azure CosmosDB NoSQL.</param>
+    /// <param name="jsonSerializerOptions">Optional JSON serializer options.</param>
+    /// <param name="options">Optional options to further configure the <see cref="IVectorStore"/>.</param>
+    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    /// <returns>The kernel builder.</returns>
+    public static IKernelBuilder AddAzureCosmosDBNoSQLVectorStore(
+        this IKernelBuilder builder,
+        string connectionString,
+        string databaseName,
+        JsonSerializerOptions? jsonSerializerOptions = default,
```

Comment:
Would it make sense to move this to the options class, and then also pass this down to the collection options?
The two have to match after all right?  The serializeroptions used by the client and the connector.  I just realised I have the same issue in Azure AI Search.  Let's chat and decide how to solve it consistently.

Reply:
That makes sense, updated!